### PR TITLE
refactoring

### PR DESCRIPTION
**Pull request by menezesd**
Describe what kind of changes you made here:

Make more use of f-strings, simplify logical expressions and control flow

dean

### CODE REVIEW

File: rpg/models.py [hunk 1]
```diff
@@ -156,7 +156,7 @@ class Meta:
         db_table = "dashboard_questcomplete"
 
     def __str__(self) -> str:
-        return self.title + " " + self.timestamp.strftime("%c")
+        return f"{self.title} " + self.timestamp.strftime("%c")
```
Comment (Incompleteusern) [inline rpg/models.py:159]: might as well fully make this an f-string

File: rpg/levelsys.py [hunk 2]
```diff
@@ -351,24 +351,23 @@ def annotate_student_queryset_with_scores(
...
     max_level = max(levels.keys())
 
     for student in annotate_student_queryset_with_scores(queryset):
-        row: dict[str, Any] = {}
-        row["student"] = student
-        row["spades"] = (getattr(student, "spades_quizzes", 0) or 0) * 2
+        row: dict[str, Any] = {
+            "student": student,
+            "spades": (getattr(student, "spades_quizzes", 0) or 0) * 2,
+        }
```
Comment (Incompleteusern) [inline rpg/levelsys.py:370]: not really a fan of this one to be honest. I'm also scared why this uses `getattr` 50,000 times.

File: rpg/levelsys.py [hunk 3]
```diff
@@ -351,24 +351,23 @@ def annotate_student_queryset_with_scores(
 
 def compute_insanity_rating(b: int, d: int, z: int) -> float:
     assert min(b, d, z) >= 0
-    if b == 0 and d == 0 and z == 0:
-        return 0
-    return (z - b) / (b + d + z)
+    return 0 if b == 0 and d == 0 and z == 0 else (z - b) / (b + d + z)
```
Comment (Incompleteusern) [inline rpg/levelsys.py:354]: I think this makes it less clear but I don't know

File: roster/views.py [hunk 4]
```diff
@@ -219,8 +221,12 @@ def master_schedule(request: HttpRequest) -> HttpResponse:
...
         unit_to_student_dicts[d["curriculum"]].append(
             {
-                "short_name": d["user__first_name"] + " " + d["user__last_name"][0:1],
-                "full_name": d["user__first_name"] + " " + d["user__last_name"],
+                "short_name": d["user__first_name"]
+                + " "
+                + d["user__last_name"][:1],
+                "full_name": d["user__first_name"]
+                + " "
+                + d["user__last_name"],
```
Comment (Incompleteusern) [inline roster/views.py:229]: this might be better as an f string

File: roster/utils.py [hunk 5]
```diff
@@ -31,10 +31,7 @@ def get_visible_from_queryset(user: User, queryset: QuerySet[models.Student]):
 
 
 def get_visible_students(user: User, current: bool = True):
-    if current:
-        queryset: QuerySet = get_current_students()
-    else:
-        queryset = models.Student.objects.all()
+    queryset = get_current_students() if current else models.Student.objects.all()
```
Comment (Incompleteusern) [inline roster/utils.py:34]: type check seems to have disappeared here.

Comment (Incompleteusern) [reply roster/utils.py:34]: I also prefer the old formatting but its fine

File: otisweb/settings.py [hunk 6]
```diff
@@ -260,9 +261,9 @@ def filter_useless_404(record: logging.LogRecord) -> bool:
         return True
     a: list[str] = [str(s) for s in record.args]
     if len(a) == 2:
-        return not (a[0] == "Not Found" and ("wp-include" in a[1] or ".php" in a[1]))
+        return a[0] != "Not Found" or "wp-include" not in a[1] and ".php" not in a[1]
     elif len(a) == 3:
-        return not (a[1] == "404" and ("wp-include" in a[0] or ".php" in a[0]))
+        return a[1] != "404" or "wp-include" not in a[0] and ".php" not in a[0]
```
Comment (Incompleteusern) [inline otisweb/settings.py:266]: This is less clear IMO.

File: markets/views.py [hunk 7]
```diff
@@ -133,14 +133,16 @@ def dispatch(
...
+            and self.market.has_ended
+        ):
             return super().dispatch(request, *args, **kwargs)
         elif not self.market.has_started:
             return HttpResponseNotFound()
-        elif not self.market.has_ended:
-            return HttpResponseRedirect(self.market.get_absolute_url())
         else:
-            return super().dispatch(request, *args, **kwargs)
+            return HttpResponseRedirect(self.market.get_absolute_url())
```
Comment (Incompleteusern) [inline markets/views.py:145]: This deletes `super().dispatch`, is this intentional?

File: markets/models.py [hunk 8]
```diff
@@ -84,14 +84,10 @@ def __str__(self) -> str:
         return f"{self.title} ({self.slug})"
 
     def get_absolute_url(self) -> str:
-        if self.has_ended:
+        if self.has_ended or not self.has_started:
```
Comment (Incompleteusern) [inline markets/models.py:87]: IMO flipping the if statement `if self.has_started and not self.has_ended` is clearer

File: arch/migrations/0013_auto_20210603_2343.py [hunk 9]
```diff
@@ -145,12 +145,12 @@
...
-    lookup[k + " RMM TST"] = v + "RST"
-    lookup[k + " EGMO TST"] = v + "EST"
-    lookup[k + " JBMO TST"] = v + "JST"
-    lookup[k + " TSTST"] = v + "TSTST"
+    lookup[f"{k} MO"] = f"{v}MO"
+    lookup[f"{k} TST"] = f"{v}TST"
+    lookup[f"{k} RMM TST"] = f"{v}RST"
+    lookup[f"{k} EGMO TST"] = f"{v}EST"
+    lookup[f"{k} JBMO TST"] = f"{v}JST"
+    lookup[f"{k} TSTST"] = f"{v}TSTST"
```
Comment (Incompleteusern) [inline arch/migrations/0013_auto_20210603_2343.py:153]: refactoring a migration seems a bit scary

File: aincrad/views.py [hunk 10]
```diff
@@ -499,7 +499,9 @@ def hanabi_handler(action: str, data: JSONData) -> JsonResponse:
                     name_to_replay_id_dict[name] = replay_json["replay_id"]
 
     for replay_json in data["replays"]:
-        if not any(name in eligible_players for name in replay_json["players"]):
+        if all(
+            name not in eligible_players for name in replay_json["players"]
+        ):
```
Comment (Incompleteusern) [inline aincrad/views.py:504]: Prefer the old one

**Review by Incompleteusern — COMMENTED** [review]
This probably should've been multiple commits.

File: arch/migrations/0013_auto_20210603_2343.py (hunk 9 above)
Comment (vEnhance) [reply arch/migrations/0013_auto_20210603_2343.py:153]: i think it's equivalent, so it should be fine... i hope...

**Comment by vEnhance** [comment]
Thanks for this.

I might wait a bit to merge this on because there's about to be a Django 4.1 -> 4.2 upgrade that might also break a lot of things, so I might want to spread out the massive changes a bit to make it easier to tell where things are breaking.

Unrelatedly, I'm curious whether you made all these changes by hand or whether there was some automation? It's a rather comprehensive PR.

**Comment by Incompleteusern** [comment]
Given that some parts are missed, I think this was done manually.

File: markets/views.py (hunk 7 above)
Comment (vEnhance) [reply markets/views.py:145]: think you need to look at all the changed lines
basically it's a DRY on the two super().dispatch cases

**Comment by vEnhance** [comment]
I'll try to merge this sometime over the coming weekend (Aug 19 - Aug 20) to allow a bit of time for the Django 4.2 migration first, just so we don't hit the server with too many changes at once.

**Comment by vEnhance** [comment]
ok it looks like the mysql/django upgrade is in working order so let's merge this now and see if anything breaks :yayy:

**Comment by vEnhance** [comment]
dean I gave you +9 spades as a thank-you for all this cleanup, much appreciated

**Pull request merged**